src/main/java/com/google/sps/data/UrlRequest.java

```diff
@@ -0,0 +1,65 @@
+package com.google.sps.data;
+
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.util.Map;
+
+/* Class conatining helper methods to make url requests */
+public final class UrlRequest {
+  /**
+   * Returns the a json string with the API response given an URL path and the query parameters.
+   */
+  public static String urlQuery(String basePath, Map<String, String> parameters) throws IOException {
+    String path = String.format("%s?%s", basePath, getParamsString(parameters));
+    URL url = new URL(path);
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
[Nit] When using acronym or initialisms in CamelCase you usually upper case only the first letter.  This variable should be called 'HttpUrlConnection'.

See https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined.

Reply:
HttpUrlConnection is a class, not a variable.